### PR TITLE
[WOC] new kudos upload / mint process

### PR DESCRIPTION
**Pull request by owocki**
##### Description
new kudos upload / mint process

##### Checklist

- [x] linter status: 100% pass
- [ ] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)

##### Affected core subsystem(s)

kudos 

##### Testing

i still need to test it. in particular:
- how do we get our gitcoin IPFS host to serve the items as an image (via the `/ipfs/` route instead of the `cat` route, and in both cases, via content type text/svg) ?
- hook up the kudos minting process to the admin web interface

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/2779

### CODE REVIEW

File: app/kudos/models.py [hunk 1]
```diff
@@ -465,3 +466,51 @@ class BulkTransferRedemption(SuperModel):
     def __str__(self):
         """Return the string representation of a model."""
         return f"coupon: {self.coupon} redeemed_by: {self.redeemed_by}"
+
+
+
+class TokenRequest(SuperModel):
+    """Define the Token model."""
```
Comment (mbeacom) [inline app/kudos/models.py]: `TokenRequest`

File: app/kudos/views.py [hunk 2]
```diff
@@ -690,3 +691,45 @@ def receive_bulk(request, secret):
...
+
+@csrf_exempt
+def newkudos(request):
+    context = {
+        'active': 'newkudos',
+        'msg': None,
+    }
+
+    if request.POST:
+        required_fields = ['name', 'description', 'priceFinney', 'artist', 'platform', 'numClonesAllowed', 'tags', 'to_address', 'artwork_url']
```
Comment (mbeacom) [inline app/kudos/views.py]: We could move this to a django form to handle this type of validation.

File: app/kudos/views.py [hunk 3]
```diff
@@ -690,3 +691,45 @@ def receive_bulk(request, secret):
...
+                platform=request.POST['platform'],
+                numClonesAllowed=request.POST['numClonesAllowed'],
+                tags=",".split(request.POST['tags']),
+                to_address=request.POST['to_address'],
+                artwork_url=request.POST['artwork_url'],
+                network='mainnet',
+                approved=False,
+                metadata={
+                    'ip': get_ip(request),
+                    'email': request.POST['email'],
```
Comment (mbeacom) [inline app/kudos/views.py]: `email` isn't validated as a required field in https://github.com/gitcoinco/web/pull/2936/files#diff-9b7a42d8513c01971adb785752bd3f67R704 - We'll probably want to use `.get('key', 'default_value')` for all of the params that aren't required here.

**Comment by owocki** [comment]
> how do we get our gitcoin IPFS host to serve the items as an image (via the /ipfs/ route instead of the cat route, and in both cases, via content type text/svg) ?

@mbeacom i could use your help with this after grants ships.  right now i can get the IPFS response to show up, but i can't get any browser to show it as an actual SVG.  i think it has to do with the content type of the reponse

File: app/kudos/models.py (hunk 1 above)
Comment (owocki) [reply app/kudos/models.py]: updated

File: app/kudos/views.py (hunk 3 above)
Comment (owocki) [reply app/kudos/views.py]: 6865ed40568af4b3397f5a91b1c4d80870d95760

**Comment by owocki** [comment]
https://github.com/gitcoinco/web/pull/2936#issuecomment-442616633

**Comment by owocki** [comment]
@mbeacom if you get a sec, i'd love to jam on this

> how do we get our gitcoin IPFS host to serve the items as an image (via the /ipfs/ route instead of the cat route, and in both cases, via content type text/svg) ?

this is the big blocker IMHO

**Comment by mbeacom** [comment]
I assume we can jam on this towards the end of the week or next week.

**Pull request closed without merging**